Sources/Auth/Services/Signer/CocoaSigner.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+protocol CacaoSignatureVerifying {
+    func verifySignature(_ cacao: Cacao) throws
+}
+
+struct CacaoSigner: CacaoSignatureVerifying {
+
+    enum Errors: Error {
+        case signatureValidationFailed
+    }
+
+    private let signer: Signer
+
+    init(signer: Signer) {
+        self.signer = signer
+    }
+
+    func sign(payload: CacaoPayload, privateKey: Data) throws -> CacaoSignature {
```

Comment:
Passing private keys around in the SDK products' layer worries me that we might be handling the keys too loosely. Can we think a way of dealing with private keys only inside the KMS module?

Reply:
Good point but client of `sign` method unimplemented yet. I suggest to think about this when it will be done